src/main/java/Duke.java

```diff
@@ -1,10 +1,240 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Duke.
+ * @author Thomas Hogben
+ */
 public class Duke {
+    private static final String MSG_ERROR_DEFAULT = "im sorry I is no understand.";
+    private static final String MSG_ERROR_BLANK_DESCRIPTION = "is no leave description blank;";
+    private static final String MSG_ERROR_INVALID_TASK_NUMBER = "what kind of number is (||❛︵❛.)";
+    private static final String MSG_ERROR_NOT_ENOUGH_TASKS = "we is dont have that many tasks yet.";
+    private static final String MSG_ERROR_UNSPECIFIED_TASK = "please is specify task please,";
+    private static final String MSG_EXIT = "okay is bye!!";
+    private static final String MSG_NO_TASKS = "is no tasks today.";
+    private static final String MSG_TASK_ADDED = "is added.";
+    private static final String MSG_TASK_DELETED = "is deleted!";
+    private static final String MSG_TASK_DONE = "is done!";
+
+    /** Program loops until this is set to false */
+    private static boolean running = true;
+    private static Scanner sc;
+    /** A list of tasks to be done */
+    private static ArrayList<Task> tasks = new ArrayList<>();
+
+    /**
+     * Lists out all Tasks numbered and on individual lines.
+     * Calls the toString() method of each Task to display them
+     * and their type/status.
+     */
+    private static void listTasks() {
+        if (tasks.size() == 0) {
+            System.out.println(MSG_NO_TASKS);
+        } else {
+            for (int i = 1; i <= tasks.size(); i++) {
+                System.out.println(i + "." + tasks.get(i - 1).toString() + ".");
+            }
+        }
+    }
+
+    /**
+     * Adds a Task to the internal list.
+     *
+     * @param newTask The new Task to be added.
+     */
+    private static void addTask(Task newTask) {
+            tasks.add(newTask);
+            System.out.println(MSG_TASK_ADDED);
```

Comment:
This is very good code quality. It is very organised and all the messages are staticly defined and that is very good.